server/controllers/BookController.js

```diff
@@ -17,25 +17,26 @@ class BookController {
    */
   static async addBook(request, response) {
     const {
-      title, description, isbn, price, yearPublished, stock, authorName
-    } = request.body;
+      body: {
+        title, description, isbn, price, yearPublished, authorName
+      }
+    } = request;
     try {
       const newAuthor = await Author.findOrCreate({ where: { fullname: authorName } });
       const existingBook = await Book.findOrCreate({
-        where: { title },
+        where: { isbn },
         defaults: {
           description,
-          isbn,
+          title,
           price,
-          yearPublished,
-          stock
+          yearPublished
         }
       });
       await BookAuthor.create({ authorId: newAuthor[0].id, bookId: existingBook[0].id });
       const { dataValues } = existingBook[0];
-      const message = `book successfully added${newAuthor[1] ? ' and author created' : ' and author exist'}`;
+      const message = `book successfully added${newAuthor[1] ? ' and author created' : ''}`;
       return existingBook[1]
-        ? responseMessage(response, 201, { status: 'success', message, book: { ...dataValues, author: authorName } })
+        ? responseMessage(response, 201, { status: 'success', message, book: { ...dataValues, author: newAuthor[0].id } })
```

Comment:
Your response here should be the author's name, not id

Reply:
Speak to biola. And I thought we were asked to change it to id by mengere

Reply:
Okay, what he meant was for data entry, imagine a drop down id would be the value, not in the response, kindly revert back so biola can create a bug fix for this